src/test/java/io/orkes/conductor/client/e2e/HttpPollWorkerTests.java

```diff
@@ -0,0 +1,177 @@
+package io.orkes.conductor.client.e2e;
+
+import com.google.common.util.concurrent.Uninterruptibles;
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.tasks.TaskDef;
+import com.netflix.conductor.common.metadata.tasks.TaskResult;
+import com.netflix.conductor.common.metadata.workflow.StartWorkflowRequest;
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+import com.netflix.conductor.common.run.Workflow;
+import io.orkes.conductor.client.ApiClient;
+import io.orkes.conductor.client.MetadataClient;
+import io.orkes.conductor.client.TaskClient;
+import io.orkes.conductor.client.WorkflowClient;
+import io.orkes.conductor.client.http.OrkesMetadataClient;
+import io.orkes.conductor.client.http.OrkesTaskClient;
+import io.orkes.conductor.client.http.OrkesWorkflowClient;
+import io.orkes.conductor.client.util.ApiUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+public class HttpPollWorkerTests {
+    static ApiClient apiClient;
+    static WorkflowClient workflowClient;
+    static MetadataClient metadataClient;
+    static TaskClient taskClient;
+
+    private final String WORKFLOW_NAME = "http_poll_wf";
+
+    @BeforeAll
+    public static void init() throws Exception {
+        apiClient = ApiUtil.getApiClientWithCredentials();
+        workflowClient = new OrkesWorkflowClient(apiClient);
+        metadataClient = new OrkesMetadataClient(apiClient);
+        taskClient = new OrkesTaskClient(apiClient);
+    }
+
+    @AfterEach
+    public void cleanup() {
+        try {
+            metadataClient.unregisterWorkflowDef(WORKFLOW_NAME, 1);
+            workflowClient.deleteWorkflow(WORKFLOW_NAME, false);
+        } catch (Exception e) {
+        }
+    }
+
+    @Test
+    public void testHttpPollTask() {
+        String url = "https://jsonplaceholder.typicode.com/posts";
```

Comment:
don't use external service.  Use orkesapi endpoint we have.  Check playground -- when you create a new http task, it automatically adds a test url - we can use that.

Reply:
Sure, let me change that to it